corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -438,7 +438,7 @@ def configure_entry_advanced_form(self, module, e, form, **kwargs):
         def case_sharing_requires_assertion(form):
             actions = form.actions.open_cases
             for action in actions:
-                if 'owner_id' in action.case_properties:
```

Comment:
if `owner_id` is specified then you actually want to not add the assertion - since it is manually set by the user.

the call below replicates the logic that automatically adds the `owner_id` in [this loop](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/xform.py#L1685).